src/assets/scripts/client/scope/RadarTargetModel.js

```diff
@@ -0,0 +1,411 @@
+     * @return {array} [success of operation, system's response]
+     */
+    moveDataBlock(commandArguments) {
+        if (_isEmpty(commandArguments)) {
+            return [false, 'ERR: BAD SYNTAX'];
+        }
+
+        let desiredDirection = commandArguments;
+        let desiredLength = '';
+
+        // FIXME: Extract this to a constants file!
+        if (commandArguments.indexOf('/') !== INVALID_NUMBER) {
+            const argumentPieces = commandArguments.split('/');
+            desiredDirection = parseInt(argumentPieces[0], DECIMAL_RADIX);
+            desiredLength = parseInt(argumentPieces[1], DECIMAL_RADIX);
+
+            if (_isEmpty(argumentPieces[0])) {
+                desiredDirection = '';
+            }
+        }
+
+        if (desiredLength > 6 || desiredLength < 0) {
+            return [false, 'ERR: LEADER LENGTH 0-6 ONLY'];
+        }
+
+        if (desiredDirection !== '' && !_isNaN(desiredDirection)) {
+            const dataBlockPositionMap = { 8: 360, 9: 45, 6: 90, 3: 135, 2: 180, 1: 225, 4: 270, 7: 315, 5: 'ctr' };
+
+            if (!_has(dataBlockPositionMap, desiredDirection)) {
+                return [false, 'ERR: BAD SYNTAX'];
+            }
+
+            this._dataBlockLeaderDirection = dataBlockPositionMap[desiredDirection];
+        }
+
+        if (desiredLength !== '' && !_isNaN(desiredLength)) {
+            this._dataBlockLeaderLength = desiredLength;
+        }
+
+        return [true, 'ADJUST DATA BLOCK'];
+    }
+
+    /**
+     * Reset all properties to their default state
+     *
+     * @for RadarTargetModel
+     * @method reset
+     */
+    reset() {
+        this.aircraftModel = null;
+        this._cruiseAltitude = INVALID_NUMBER;
+        this._dataBlockLeaderDirection = INVALID_NUMBER;
+        this._dataBlockLeaderLength = this._theme.DATA_BLOCK.LEADER_LENGTH;
+        this._hasFullDataBlock = true;
+        this._hasHalo = false;
+        this._hasSuppressedDataBlock = false;
+        this._interimAltitude = INVALID_NUMBER;
+        this._isUnderOurControl = true;
+        this._routeString = '';
+        this._scratchPadText = '';
+    }
+
+    /**
+     * Set the value of the scratchpad
+     *
+     * @for RadarTargetModel
+     * @method setScratchpad
+     * @param scratchPadText {string}
+     * @return {array} [success of operation, system's response]
+     */
+    setScratchpad(scratchPadText) {
+        this._scratchPadText = scratchPadText;
+
+        return [true, 'SET SCRATCHPAD'];
+    }
+
+    /**
+     * Toggle halo (circle) on and off
+     *
+     * @for RadarTargetModel
+     * @method toggleHalo
+     * @return {array} [success of operation, system's response]
+     */
+    toggleHalo() {
+        this._hasHalo = !this._hasHalo;
+
+        return [true, 'TOGGLE HALO'];
+    }
+
+    /**
+     * Initialize the value of the scratchpad
+     *
+     * @for RadarTargetModel
+     * @method _initializeScratchPad
+     */
+    _initializeScratchPad() {
```

Comment:
move to after `_init()` with other lifecycle methods defined in calling order

Reply:
done